pylint/checkers/deprecated.py

```diff
@@ -26,15 +28,65 @@ class DeprecatedMixin(metaclass=abc.ABCMeta):
             "deprecated-method",
             "The method is marked as deprecated and will be removed in the future.",
         ),
+        "W1511": (
+            "Using deprecated argument %s of method %s()",
+            "deprecated-argument",
+            "The argument is marked as deprecated and will be removed in the future.",
+        ),
     }
 
-    @abc.abstractmethod
+    @utils.check_messages(
+        "deprecated-method",
+        "deprecated-argument",
+    )
+    def visit_call(self, node):
+        """Called when a :class:`.astroid.node_classes.Call` node is visited.
+
+        Args:
+            node (astroid.node_classes.Call): The node to check.
+        """
+        try:
+            for inferred in node.func.infer():
+                # Calling entry point for deprecation check logic.
+                self.check_deprecated_method(node, inferred)
+        except astroid.InferenceError:
+            return
+
     def deprecated_methods(self):
         """Callback returning the deprecated methods/functions.
 
         Returns:
             collections.abc.Container of deprecated function/method names.
         """
+        # pylint: disable=no-self-use
+        return ()
+
+    def deprecated_arguments(self, method: str):
```

Comment:
I like this, it will make the code cleaner and it's the same mechanism than ``deprecated_methods``, that's pretty clever.